cpp/src/arrow/util/bit_util.cc

```diff
@@ -598,5 +598,49 @@ Result<std::shared_ptr<Buffer>> BitmapAllButOne(MemoryPool* pool, int64_t length
   return std::move(buffer);
 }
 
+std::pair<int64_t, int64_t> BitmapScanner::NextRun() {
+  auto load_word = [](const uint8_t* bytes) -> uint64_t {
+    return BitUtil::ToLittleEndian(util::SafeLoadAs<uint64_t>(bytes));
+  };
+  auto shift_word = [](uint64_t current, uint64_t next, int64_t shift) -> uint64_t {
+    return (current >> shift) | (next << (64 - shift));
```

Comment:
will this have different semantics on big endian?

Reply:
This `shift_word` helper is used in other functions -- the unit tests pass on big endian it seems

Reply:
I added some more tests with random data